apps/web/app/(roles)/vagas/page.tsx

```diff
@@ -1,63 +1,76 @@
+'use server'
+
 import { RolesPage } from './RolesPage'
 import { createClient } from '@supabase/supabase-js'
 import { createClient as createClientRedis } from 'redis'
 import { fetchJobs } from './action'
 
-export const revalidate = 0
-
 const ONE_DAY_IN_MINUTES = 86_400
 
-const client = createClientRedis({
-  socket: {
-    host: process.env['REDIS_HOST'],
-    port: parseInt(process.env['REDIS_PORT'] || '6379'),
-  },
-})
-
-client.connect()
+async function getRedisClient() {
+  const client = createClientRedis({
+    socket: {
+      host: process.env['REDIS_HOST'],
+      port: parseInt(process.env['REDIS_PORT'] || '6379'),
+    },
+  })
+  await client.connect()
+  return client
+}
 
-const supabase = createClient(
-  process.env['SUPABASE_URL'] as string,
-  process.env['SUPABASE_SERVICE_ROLE'] as string
-)
+function getSupabaseClient() {
+  return createClient(
+    process.env['SUPABASE_URL'] as string,
+    process.env['SUPABASE_SERVICE_ROLE'] as string
+  )
+}
```

Comment:
Usually we keep this type of code separated from the UI page, so it can be re-used in other places of the application.

If i'm not mistaken, there's already a function in shared-ui that handles this, please use it.

Reply:
I haven't found the getSupabase function in shared, so I created it in the utils of web. Now we have a function to getRedisClient and getSupabaseClient